sqlalchemy_exasol/base.py

```diff
@@ -828,33 +872,41 @@ def get_pk_constraint(self, connection, table_name, schema=None, **kw):
     #        cursor=foreignKeys_cursor.foreignKeys()
     #        return [(row.fk_name, row.fkcolumn_name, row.pktable_schema, row.pktable_name, row.pkcolumn_name, fktable_name, "FOREIGN KEY") for row in cursor]
 
+
     @reflection.cache
-    def _get_foreign_keys_odbc(self, connection, odbc_connection, schema=None, **kw):
+    def _get_foreign_keys_odbc(self, connection, odbc_connection, table_name, schema=None, **kw):
         # Need to use a workaround, because SQLForeignKeys functions doesn't work for an unknown reason
         tables = self._get_tables_for_schema_odbc(connection=connection, odbc_connection=odbc_connection,
-                                                  schema=schema, table_type="TABLE")
+                                                  schema=schema, table_name=table_name, table_type="TABLE", **kw)
         if len(tables) > 0:
-            quoted_schema_string = self.quote_string_value(tables[0][1])
+            quoted_schema_string = self.quote_string_value(tables[0].table_schem)
+            quoted_table_string = self.quote_string_value(tables[0].table_name)
             sql_stmnt = \
-                "/*snapshot execution*/ " \
-                "SELECT constraint_name, column_name, referenced_schema, referenced_table, " \
-                "referenced_column, constraint_table, constraint_type " \
-                "FROM SYS.EXA_ALL_CONSTRAINT_COLUMNS " \
-                "WHERE constraint_schema={schema} and constraint_type='FOREIGN KEY' " \
-                "ORDER BY ordinal_position" \
-                    .format(schema=quoted_schema_string)
-            rp = connection.execute(sql.text(sql_stmnt),
-                                    schema=self.denormalize_name(schema))
+                "/*snapshot execution*/ " + \
+                self._get_constraint_sql_str(quoted_schema_string,quoted_table_string,"FOREIGN KEY")
+            rp = connection.execute(sql.text(sql_stmnt))
             return list(rp)
         else:
             return []
 
     @reflection.cache
-    def _get_foreign_keys_sql(self, connection, schema=None, **kw):
-        return self._get_all_constraints(connection, schema=schema, info_cache=kw.get("info_cache"))
+    def _get_foreign_keys_sql(self, connection, table_name, schema=None, **kw):
+        table_name_string = ":table"
+        if schema is None:
+            schema_string = "CURRENT_SCHEMA "
+        else:
+            schema_string = ":schema "
+        sql_stmnt = \
+            self._get_constraint_sql_str(schema_string, table_name_string, "FOREIGN KEY")
+        rp = connection.execute(sql.text(sql_stmnt),
+                                schema=self.denormalize_name(schema),
+                                table=table_name)
+        return list(rp)
 
     @reflection.cache
     def get_foreign_keys(self, connection, table_name, schema=None, **kw):
+        if table_name is None:
+            raise RuntimeError("table_name must not null")
```

Comment:
Was this previously an uncaught exception or the exception didn't happen in the first place?

Reply:
Previously, we returned foreign keys for all tables, however the interface documentation of sqlalchemy doesn't describe the case. As an alternative, we could return None or something like that.

  https://docs.sqlalchemy.org/en/13/core/internals.html?highlight=get_columns#sqlalchemy.engine.interfaces.Dialect.get_pk_constraint